canvas/models/models.py

```diff
@@ -175,9 +175,10 @@ def username(self):
 
     @property
     def name(self):
-        if self.user.has_name:
-            return self.user.get_full_name()
-        return "Anonymous Student"
+        if not self.user.nickname:
+            raise ValueError(f"User {self.user.id} does not have a nickname")
```

Comment:
This is not a good practice, we should not raise error. If the nickname field is `NOT NULL` in the database it will always have value.